test/instrumentation/modules/@aws-sdk/client-sqs.test.js

```diff
@@ -0,0 +1,306 @@
+      );
+      t.equal(
+        spans.filter((s) => s.transaction_id === tx.id).length,
+        spans.length,
+        'all spans have the same transaction_id',
+      );
+      t.equal(
+        spans.filter((s) => s.sync === false).length,
+        spans.length,
+        'all spans have sync=false',
+      );
+      t.equal(
+        spans.filter((s) => s.sample_rate === 1).length,
+        spans.length,
+        'all spans have sample_rate=1',
+      );
+
+      // Keep IDs for link assertions
+      const sendMessageSpanId = spans[0].id;
+      const sendMessagesBatchSpanId = spans[1].id;
+
+      spans.forEach((s) => {
+        // Remove variable and common fields to facilitate t.deepEqual below.
+        delete s.id;
+        delete s.transaction_id;
+        delete s.parent_id;
+        delete s.trace_id;
+        delete s.timestamp;
+        delete s.duration;
+        delete s.sync;
+        delete s.sample_rate;
+      });
+
+      t.deepEqual(
+        spans.shift(),
+        {
+          name: 'SQS SEND to elasticapmtest-queue-1.fifo',
+          type: 'messaging',
+          subtype: 'sqs',
+          action: 'send',
+          context: {
+            service: {
+              target: { type: 'sqs', name: 'elasticapmtest-queue-1.fifo' },
+            },
+            destination: {
+              address: LOCALSTACK_HOST,
+              port: 4566,
+              cloud: { region: 'us-east-2' },
+              service: {
+                type: '',
+                name: '',
+                resource: 'sqs/elasticapmtest-queue-1.fifo',
+              },
+            },
+            message: { queue: { name: 'elasticapmtest-queue-1.fifo' } },
+          },
+          outcome: 'success',
+        },
+        'sendMessage',
+      );
+
+      t.deepEqual(
+        spans.shift(),
+        {
+          name: 'SQS SEND_BATCH to elasticapmtest-queue-1.fifo',
+          type: 'messaging',
+          subtype: 'sqs',
+          action: 'send_batch',
+          context: {
+            service: {
+              target: { type: 'sqs', name: 'elasticapmtest-queue-1.fifo' },
+            },
+            destination: {
+              address: LOCALSTACK_HOST,
+              port: 4566,
+              cloud: { region: 'us-east-2' },
+              service: {
+                type: '',
+                name: '',
+                resource: 'sqs/elasticapmtest-queue-1.fifo',
+              },
+            },
+            message: { queue: { name: 'elasticapmtest-queue-1.fifo' } },
+          },
+          outcome: 'success',
+        },
+        'sendMessageBatch',
+      );
+
+      // There will be one or more `SQS POLL ...` spans for the ReceiveMessage
+      // API calls until all messages are retrieved -- with interspersed
+      // `SQS DELETE_BATCH ...` spans to delete those messages as they are
+      // received.
+      const spanLinks = [];
+
+      while (spans.length > 0) {
+        const currSpan = spans.shift();
```

Comment:
nit: If the `break` case is hit below -- i.e. `spans` includes some span other than `SQS POLL ...` or `SQS DELETE_BATCH ...`, e.g. if `use-client-sqs.js` is testing some other SQS commands -- then this `spans.shift()` will have removed that span from `spans` for subsequent testing.  This is why the while-loop in "sqs.test.js" isn't `.shift()`ing until `topSpanName` has been identified as one we want to process.

Just a nit though because, it works here because `spans` is empty after the `POLL/DELETE_BATCH` spans.

Reply:
You're right. I've interpreted the condition of the `while` as "let's consume all the remaining spans".